pkg/cmd/release/create/create.go

```diff
@@ -92,9 +95,16 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 
 			Upload a release asset with a display label
 			$ gh release create v1.2.3 '/path/to/asset.zip#My display label'
+
+			Create a release and start a discussion
+			$ gh release create v1.2.3 --discussion-category "General"
 		`),
 		Args: cmdutil.MinimumArgs(1, "could not create: no tag name provided"),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Flags().Changed("discussion-category") && opts.Draft {
+				return errors.New("Discussions for draft releases not supported")
```

Comment:
Will the server also error out on this? If yes, I would propose removing the client-side validation. Maybe in the future the platform allows this (e.g. saves the parameter in a draft release and uses it upon publish), and in that case this bit of client-side validation will be unwanted.

Reply:
Just tested this. The server creates the release and ignores the discussion category. No message or anything from the server that the discussion category is not allowed for draft releases. Maybe worth relaying that to the backend team or w/e as that might be confusing, especially if you're trying to automate this stuff. I'll leave in this validation and push changes for the rest of your comments. Look forward to hearing your feedback.

Reply:
It's fine to leave the validation for now. I don't think it's extremely likely that the platform will accept and respect this parameter for draft releases anytime soon.